DirectProgramming/DPC++FPGA/Tutorials/Features/dsp_control/README.md

```diff
@@ -0,0 +1,192 @@
+
+### Scope of Datatypes and math operations
+| Datatypes              | math operations
+---                      |---
+| `float`                | Add, sub, constant mul
+| `hls_float<8, 23>`     | Add, sub, constant mul
+| `int`                  | Constant mul
+| `ac_int`               | Constant mul
+| `ac_fixed`             | Constant mul
+
+**NOTE:** _constant mul_ means one operand of the multiplication is a constant.
+
+### Global Control
+The `-Xsdsp-mode=<option>` command-line flag sets implementation preference of math operations that support DSP control in all kernels. It has three valid options:
+| Option               | Explanation
+---                    |---
+| `default`            | This is the default option if this command-line flag is not passed manually. The compiler determines the implementation based on datatype and math operation.
+| `prefer-dsp`         | Prefer math operations to be implemented in **DSPs**. If a math operation is implemented by DSPs by default, you will see no difference in resource utilization or area. Otherwise, you will notice a decrease in the usage of soft-logic resources and an increase in the usage of DSPs.
+| `prefer-softlogic`   | Prefer math operations to be implemented in **soft-logic**. If a math operation is implemented without DSPs by default, you will see no difference in resource utilization or area. Otherwise, you will notice a decrease in the usage of DSPs and an increase in the usage of soft-logic resources.
+
+### Local Control
+The library function `math_dsp_control<Preference::<enum>, Propagate::<enum>>([&]{})` provides block scope local control in one single kernel. A reference-capturing lambda expression is passed as the argument to this library function. Inside the lambda expression, implementation preference of math operations that support DSP control will be determined by two template arguments.
+
+The first template argument `Preference::<option>` is an enum with three valid options:
+| Option               | Explanation
+---                    |---
+| `DSP`                | Prefer math operations to be implemented in **DSPs**. Its behavior on a math operation is equivalent to global control `-Xsdsp-mode=prefer-dsp`. <br> **NOTE:** This option will be automatically applied if the template argument `Preference` is not specified manually.
+| `Softlogic`          | Prefer math operations to be implemented in **soft-logic**. Its behavior on a math operation is equivalent to global control `-Xsdsp-mode=prefer-softlogic`.
+| `Compiler_default`   | Compiler determines the implementation based on datatype and math operation. Its behavior on a math operation is equivalent to global control `-Xsdsp-mode=default`.
+
+The second template argument `Propagate::<option>` is an enum that determines propagation of the first argument `Preference` to functions called in the lambda expression:
```

Comment:
This tutorial does not explain this concept in a way that I understand.  Perhaps the sample code needs to demonstrate the use of this Propogate argument?  After reading this, I don't get it.

Reply:
I'll add some illustrations in the README.